Sources/InkCLI/main.swift

```diff
@@ -7,22 +7,51 @@
 import Foundation
 import Ink
 
-guard CommandLine.arguments.count > 1 else {
-    print("""
-    Ink: Markdown -> HTML converter
-    -------------------------------
-    Pass a Markdown string to convert as input,
-    and HTML will be returned as output. To use
-    STDIN as input, call ink with "-" as a single
-    argument, like this: '$ ink -'.
-    """)
+if CommandLine.arguments.contains(where: { $0 == "-h" || $0 == "--help" }) {
+    print(helpMessage)
     exit(0)
 }
 
-var markdown = CommandLine.arguments[1]
+let markdown: String
+let arguments = CommandLine.arguments
 
-if markdown == "-" {
+switch arguments.count {
+case 1:
+    // no arguments, parse stdin
     markdown = AnyIterator { readLine() }.joined(separator: "\n")
+case let count where arguments[1] == "-m" || arguments[1] == "--markdown":
+    // first argument -m or --markdown, parse Markdown string
+    guard count == 3 else {
+        printError("-m, --markdown flag takes a single following argument")
+        printError(usageMessage)
+        exit(1)
+    }
+    markdown = CommandLine.arguments[2]
+case 2:
+    // single argument, parse contents of file
+    let fileUrl: URL
+    if CommandLine.arguments[1].hasPrefix("/") {
```

Comment:
Maybe we should also check for `~` and replace it with a home directory path?

Reply:
Great idea! There's an `NSString` API for that:
```swift
NSString(string: CommandLine.arguments[1]).expandingTildeInPath
```
And since we depend on Foundation for the CLI, it's no problem doing that bridging IMO.

Reply:
Good idea. Bash was expanding ~ to an absolute string, so `ink ~/file.md` would parse correctly through the CLI's absolute path, but `ink "~/file.md"` would look for `/$PWD/~/file.md`. Doesn't hurt to add a case for this.

Reply:
Really great attention to detail 👍